controllers/topology/internal/mergepatch/mergepatch.go

```diff
@@ -259,11 +277,24 @@ func enforcePath(authoritative, twoWay map[string]interface{}, path contract.Pat
 
 	default:
 		// If in the middle of a path, go into the nested map,
-		nestedSimpleMap, ok := authoritative[path[0]].(map[string]interface{})
+		var nestedSimpleMap map[string]interface{}
+		switch v, ok := authoritative[path[0]]; {
+		case ok && v == nil:
+			// If the nested map is nil, it means that the authoritative patch is trying to delete a parent object
+			// in the middle of the enforced path.
+
+			// If the parent object has been intentionally deleted (the corresponding parent object value in the modified object is null),
+			// then we should enforce the deletion of the parent object including everything below it.
+			if _, ok := modified[path[0]]; ok && v == nil {
```

Comment:
v == nil is always true as we check it in l.282

Reply:
fixed, this should be m (the value from modified, not v)

Reply:
Thought so